client/chain_client.go

```diff
@@ -34,44 +34,51 @@ type ChainClient struct {
 }
 
 func NewChainClient(ccc *ChainClientConfig, input io.Reader, output io.Writer, kro ...keyring.Option) (*ChainClient, error) {
-	if err := ccc.Validate(); err != nil {
-		return nil, err
-	}
-
 	// TODO: test key directory and return error if not created
 	keybase, err := keyring.New(ccc.ChainID, ccc.KeyringBackend, ccc.KeyDirectory, input, kro...)
```

Comment:
should we move this keybase down into init?

Reply:
the reason it's not in there is because on the provider interface we take no function parameters to init so then `input` and `kro` can't be passed in